core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java

```diff
@@ -506,10 +506,14 @@ public Node visitUpdateAssignment(SqlBaseParser.UpdateAssignmentContext context)
     @Override
     public Node visitMerge(SqlBaseParser.MergeContext context)
     {
+        Table table = new Table(getLocation(context), getQualifiedName(context.qualifiedName()));
+        Relation targetRelation = table;
+        if (context.identifier() != null && !context.identifier().isEmpty()) {
```

Comment:
`&& !context.identifier().isEmpty()` seems redundant.

Reply:
I wasn't sure of the contract.  Removed.